File: Keepr/Models/Keep.cs

namespace Keepr.Models;

public class Keep
{
    public int Id { get; set; }
    public string CreatorId { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public int Views { get; set; }
    public string Img { get; set; }
    public int Kept { get; set; }
    public Profile Creator { get; set; }

}
// Kept is || reportCount inHelpReviews lecture
// virtual properties are returned from tables, not on the table itself

File: Keepr.client/src/models/Keep.js
import { Profile } from "./Account.js"
export class Keep {
    constructor(data)
    {
        this.id = data.id
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.kept = data.kept
        this.createdAt = data.createdAt
        this.creatorId = data.creatorId
        this.creator = new Profile(data.creator)
    }
}